typespec-extension/src/code-model-builder.ts

```diff
@@ -564,108 +517,148 @@ export class CodeModelBuilder {
         codeModelClient.apiVersions = [];
         for (const version of this.getFilteredApiVersions(
           this.apiVersion,
-          versioning.getVersions(),
+          versions,
           this.options["service-version-exclude-preview"],
         )) {
           const apiVersion = new ApiVersion();
-          apiVersion.version = version.value;
+          apiVersion.version = version;
           codeModelClient.apiVersions.push(apiVersion);
         }
       }
 
-      // server
+      // client initialization
       let baseUri = "{endpoint}";
-      const servers = getServers(this.program, client.service);
-      if (servers && servers.length === 1 && !this.isArmSynthesizedServer(servers[0])) {
-        baseUri = servers[0].url;
-      }
-      const hostParameters = this.processHost(servers?.length === 1 ? servers[0] : undefined);
-      codeModelClient.addGlobalParameters(hostParameters);
+      let hostParameters: Parameter[] = [];
+      client.initialization.properties.forEach((initializationProperty) => {
+        if (initializationProperty.kind === "endpoint") {
+          let sdkPathParameters: SdkPathParameter[] = [];
+          if (initializationProperty.type.kind === "union") {
+            if (initializationProperty.type.values.length <= 2) {
+              // only get the path parameters from the endpoint whose serverUrl is not {"endpoint"}
+              for (const endpointType of initializationProperty.type.values) {
+                if (endpointType.kind === "endpoint" && endpointType.serverUrl !== "{endpoint}") {
+                  sdkPathParameters = endpointType.templateArguments;
+                  baseUri = endpointType.serverUrl;
+                }
+              }
+            } else {
+              throw new Error("multiple server url defined for one client is not supported yet");
```

Comment:
Let us be more consistent on error message. Make it a whole sentence, start with uppercase letter and end with `.`.